README.md

```diff
@@ -14,30 +14,36 @@ an exact solution if possible or use an appropriate numeric solver otherwise.
 
 `src` - The source code of NESTML
 
-## Docker
-The docker files can be find in the `docker` folder.
+## Running NESTML using Docker
 
-For the usage:
-Docker must be installed on the machine. See the follwing resource for the installation instructions https://docs.docker.com/engine/installation
-Clone the github repository locally. Switch to the docker-folder in the nestml repository.
+NESTML has quite some dependencies, which makes it a bit complicated to install and run it. To lower the burden, we have created a [Docker](https://www.docker.com/) container for you. The `Dockerfile`s and corresponding helper scripts can be found in the `docker` folder. In order to use this method, you have to have Docker installed on your machine. Please refer to the [installation instructions](https://docs.docker.com/engine/installation) or use the packages from your Linux distribution's software manager.
+
+### Provisioning
+
+The container can be provisioned (created) by changing to the `docker` directory of your clone of the `nestml` Git repository and running
 
 ```
-sh nestml_docker.sh provision
+./nestml_container.sh provision
 ```
-If everything goes well, then command 'docker images' container should list a 'nestml_release' container.
 
-Currently, the container is built in the way, that is automatically processes a volume that is mounted at start up. E.g. a NESTML model (https://github.com/nest/nestml/blob/master/models/iaf_cond_alpha_implicit.nestml) can be placed into a folder called 'testing' in the home-folder. Then, the following command will execute model analysis and codegeneration for NEST target (currently, only the NEST master is supported, for earlier version use releases prio 0.1.0):
+This will download all required packages and libraries and create a container that uses the pre-built version of the [latest release of NESTML](https://github.com/nest/nestml/releases). If you are interested in using the bleeding edge version of NESTML (a.k.a. Git master), you can add the argument `--dev` to the invocation of the `./nestml_container.sh` script.
+
+If everything goes well, the list printed by the command 'docker images' should now contain the 'nestml_release' container. If you experience an error, please [open an issue](https://github.com/nest/nestml/issues) so we can look into and fix it.
+
+### Running
+
+To actually convert your model files written in NESTML to NEST C++, you have to run the Docker container. This is again done using the `nestml_container.sh` script, which for this purpos gets the command `run` as first argument and one or more folders containing one or more `.nestml` files (the folder is called `<models>` in the following description):
+
 ```
-sh nestml_docker.sh run ~/testing
+./nestml_container.sh run <models>
```

Comment:
I agree with @DimitriPlotnikov. The Docker container will always put generated files into a (new or existing) directory `build` below the `<models>` directory.

However, in 50f103de71d737 I've added a section about installing and running the nestml tool as you suggest above. There, I also mention the `--target` option.

@DimitriPlotnikov: the term `nestml_container.sh` was used in error instead of the proper name `nestml_docker.sh`. I've corrected this in f2d27e51f96e9